src/components/store/ItemCard/index.tsx

```diff
@@ -26,25 +27,34 @@ type ItemCardProps = (StoreItemOptions | CollectionOptions) & CommonOptions;
  * - a store item that costs `cost` diamonds (optional `outOfStock`), or
  * - a collection with a `description`.
  */
-const ItemCard = ({ image, title, href, className, ...props }: ItemCardProps) => {
-  const Card = href ? Link : 'article';
+const ItemCard = ({
+  image,
+  title,
+  href,
+  className,
+  children,
+  ...props
+}: PropsWithChildren<ItemCardProps>) => {
   return (
-    <Card href={href ?? ''} className={`${styles.itemCard} ${className}`}>
-      <div className={styles.imageWrapper}>
-        <Image src={image} alt={title} fill />
-      </div>
-      <div className={styles.details}>
-        <p className={styles.title}>{title}</p>
-        {'description' in props && <p>{props.description}</p>}
-        {'cost' in props && (
-          <p className={styles.cost}>
-            <Diamonds count={props.cost} />
-            &nbsp;
-            {props.outOfStock && <span className={styles.outOfStock}>Out of stock</span>}
-          </p>
-        )}
-      </div>
-    </Card>
+    <article className={`${styles.itemCard} ${className}`}>
+      <Link href={href} className={styles.linkWrapper}>
+        <div className={styles.imageWrapper}>
+          <Image src={image} alt={title} fill />
+        </div>
+        <div className={styles.details}>
+          <p className={styles.title}>{title}</p>
+          {'description' in props && <p>{props.description}</p>}
+          {'cost' in props && (
+            <p className={styles.cost}>
+              <Diamonds count={props.cost} />
```

Comment:
Can we also add the discount value here on Diamonds? I've noticed it's not being displayed when it should